plugins/techdocs-node/src/stages/generate/helpers.ts

```diff
@@ -150,31 +150,80 @@ export const MKDOCS_SCHEMA = DEFAULT_SCHEMA.extend([
   }),
 ]);
 
+/**
+ * Generates a mkdocs.yml configuration file
+ *
+ * @param inputDir - base dir to where the mkdocs.yml file will be created
+ * @param siteName - name of site to be used in mkdocs.yml for the
+ * required `site_name` property, default value is "Table of Contents"
+ */
+export const generateMkdocsYml = async (
+  inputDir: string,
+  siteName?: string,
+) => {
+  try {
+    // TODO(awanlin): Use a provided default mkdocs.yml
+    // from config or some specified location. If this is
+    // not provided then fall back to generating bare
+    // minimum mkdocs.yml file
+
+    const mkdocsYmlPath = path.join(inputDir, 'mkdocs.yml');
+    const defaultSiteName = siteName ?? 'Table of Contents';
+    const defaultMkdocsContent =
```

Comment:
Would it make sense to create an object that stringify with `YAML`?

Reply:
Yeah, makes sense to me, will make other things easier down the road too if I'm thinking about this properly. Done!